File: factorial.py
print('Factorial of a number')
f =1
i=input('Enter a number(n) to get its factorial ' )
try:
	x=int(i)
	if x >=0:
		for y in range(1,x+1):
			f=f*y
		print(f)
	else:
		print('This is a negative integer')
except ValueError:
	print("The input value is not an integer")

File: factorial.cxx
#include<iostream>
using namespace std;
int main()
{
cout<<"The factorial of a whole number"<<endl<<endl;
cout<<"Enter the number(n) to get its factorial: ";
int a, b, n=1;
cin>>b;
if (b>=0){
	for(a=1; a<=b; a++){
		n=n*a;
	}	
cout<<"n! = "<<n<<endl;	
}
else cout<<"It is not a whole number!"<<endl;
}